src/main/java/io/github/opencubicchunks/cubicchunks/world/storage/CubeSerializer.java

```diff
@@ -256,8 +258,9 @@ public static CompoundTag write(ServerLevel world, IBigCube icube) {
         for (int i = 0; i < IBigCube.SECTION_COUNT; ++i) {
             LevelChunkSection section = sections[i];
 
-            DataLayer blockData = worldlightmanager.getLayerListener(LightLayer.BLOCK).getDataLayerData(Coords.sectionPosByIndex(pos, i));
-            DataLayer skyData = worldlightmanager.getLayerListener(LightLayer.SKY).getDataLayerData(Coords.sectionPosByIndex(pos, i));
+            DataLayer blockData = data != null ? data.blockLight.get(Coords.sectionPosByIndex(pos, i)) :
+                worldlightmanager.getLayerListener(LightLayer.BLOCK).getDataLayerData(Coords.sectionPosByIndex(pos, i));
+            DataLayer skyData = data != null ? data.skyLight.get(Coords.sectionPosByIndex(pos, i)) : worldlightmanager.getLayerListener(LightLayer.SKY).getDataLayerData(Coords.sectionPosByIndex(pos, i));
```

Comment:
Not sure why you've included null checks.

Reply:
Because some code actually supplies null here, look at other parts of code

Reply:
We don't need `AsyncSaveData` when saving on server thread. 
https://github.com/OpenCubicChunks/CubicChunks/pull/688/files#diff-15d7a01f28f739e39679f2ffa457058907df1e9beba5c9e5ab71aba855ae2b95R353